File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seen = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(seen.containsKey(target - nums[i])){
                int[] arr = new int[2];
                arr[0] = seen.get(target - nums[i]);
                arr[1] = i;
                return arr;
            }else{
                seen.put(nums[i], i);
            }
        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        middle = (r + l) // 2
        while l <= r:
            if nums[middle] < target:
                l = middle + 1
                middle = (r + l) // 2
            elif nums[middle] > target:
                r = middle - 1
                middle = (r + l) // 2
            elif target == nums[middle]:
                return middle
        return -1